sycl/plugins/unified_runtime/ur/adapters/cuda/image.cpp

```diff
@@ -383,8 +383,8 @@ UR_APIEXPORT ur_result_t UR_APICALL urBindlessImagesUnsampledImageCreateExp(
     UR_CHECK_ERROR(cuSurfObjectCreate(&surface, &image_res_desc));
     *phImage = (ur_exp_image_handle_t)surface;
 
-    auto urMemObj = std::unique_ptr<ur_mem_handle_t_>(new ur_mem_handle_t_{
-        hContext, (CUarray)hImageMem, surface, pImageDesc->type});
+    auto urMemObj = std::unique_ptr<ur_mem_handle_t_>(
+        new ur_image_{hContext, (CUarray)hImageMem, surface, pImageDesc->type});
```

Comment:
Where is `ur_image_` coming from? And why is the unique ptr template type a `ur_mem_handle_t_` instead of the contained type `ur_image_`?

Reply:
`ur_image_` is a derived class of `ur_mem_handle_t`. Multi dev ctx requires this to change since the current design, using unions, cannot hold dynamically allocated objects like `std::vector`s, which are needed for multi dev ctx to work.

We are using the `unique_ptr<ur_mem_handle_t>` so that we can use the `ur_image_`s and `ur_buffer_`s polymorphically, through ptrs to the base class

Reply:
Ah, I see. Thanks!

Reply:
This PR is a bit stale as https://github.com/intel/llvm/pull/11105 is more important. I will be revisiting this PR once the other has merged and I will coordinate with your team to say what it is doing, and ask for thoughts or suggestions.